README.md

```diff
@@ -1,18 +1,120 @@
 
-The project is tested using [hardhat](https://hardhat.org/), the [starknet
-hardhat plugin](https://github.com/Shard-Labs/starknet-hardhat-plugin) and
-[starknet-devnet](https://github.com/Shard-Labs/starknet-devnet).
+## Contracts
+
+`L1`
+  * `StaticATokenLM` - an updated implementation of staticATokens which makes it possible to communicate with its respective L2 token by sending the latest `rewards_index` on token transfer or when explicitly triggered to do so.
+  *  `TokenBridge` -  handles rewards update, deposit & withdrawal of staticATokens & their underlying assets
+  *  `Proxy` - A proxy implementation 
+
+`L2`
+  * `ETHStaticAToken` - Tokens on Starknet equivalent to each staticAToken on Ethereum mainnet. Contains the same logic for tracking user rewards as staticATokens and has the same `rewards_index`.
+  * `claimable` - tracks users' claimable rewards and current reward index for each `ETHStaticAToken`
+  * `token_bridge` - is responsible for:
+    * bridging the staticATokens to and from L2. Minting and burning ETHStaticATokens on message from L1. 
+    * bridging rewAAVE token back to L1
+    * updating `rewards_index` for each ETHStaticAToken on message from L1 
+  * `rewAAVE` - an ERC20 representing the rewards on L2
+  *  `proxy` - a generic implementation of a proxy in starknet
+
+## ETHStaticATokens on L2
+
+Natively, on Aave aTokens grow in balance, not in value. To be able to create this kind of model, it is important to wrap them before bridging, converting them in a token that grows in value, not in balance.
+
+ETHStaticATokens are an implementation of the wrapped aTokens that will continuously increase in value on Starknet because they are backed by the increasing staticATokens amounts locked in the bridge contract on Ethereum. The ETHStaticATokens can then be bridged back to staticATokens.
+
+## Bridging staticATokens from L1<>L2
+
+### Transfer L1->L2: 
+
+Users can either bridge their staticAToken to L2 by calling `deposit()` on `TokenBridge`, or deposit the underlying asset of the staticAToken directly by calling `depositUnderlying()`.
+
+Calling `deposit` will result in the following:
+
+- staticATokens will be transfered from the user account to the L1 bridge
+- A `deposit` event will be emitted with the L1 token address, the recipient address on L2, and the amount
+- A message will be sent to the  L2 bridge with the amount to be transferred, the L1 token address and the recipient address as parameters.
+- The token bridge on L2 will then be minting the correspending ETHStaticAToken of the L1 token to the user.
+
+Calling `depositUnderlying` will result in the following:
+
+Users will have to approve the bridge to spend the underlying `asset` tokens or `aTokens`, depending on the provided value for `fromAsset` argument when depositing.
+
+- the underlying asset of the staticAToken or the aToken (depending on the user's input )  will be transfered from the user account to the L1 bridge
```

Comment:
sorry for the format comment but sentences should start with a capital character and end with a dot.